### PR TITLE
refactor(x/data): update query service and commands

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #714 / #1152

This pull request makes the following changes:

- updates `ByIRI` and `ByHash` to `AnchorByIRI` and `AnchorByHash` for clarification
- updates `AttestorsByIRI` and `AttestorsByHash` to `AttestationsByIRI` and `AttestationsByHash` for clarification
- updates `ContentEntry` to `AnchorInfo` and moves to `query.go` for consistent query response type naming
- updates `AttestorEntry` to `AttestationInfo` and moves to `query.go` for consistent query response type naming
- updates conversion queries to not use the `Query` prefix for clarification (we are not querying state)
- removes conversion queries by the properties of each hash type and add `post` option for `ConvertHashToIRI`
- updates all other queries `ByHash` to include `post` option providing previously excluded gRPC REST endpoints
- updates comments and tests to reflect the changes listed above and cleans up integration tests

If need be, I can break this up into smaller pull requests but one thing led to another...

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
Ready for review again. I can walk through the changes and break out into small pull requests if needed.

File: x/data/server/query_attestations_by_attestor.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidAddress.Wrapf("attestor: %s", err)
+	}
+
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := s.stateStore.DataAttestorTable().List(
```
Comment (aleem1314) [inline x/data/server/query_attestations_by_attestor.go:32]: Let's also close the iterator.

File: x/data/server/query_attestations_by_hash.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return nil, sdkerrors.ErrNotFound.Wrap("data record with content hash")
+	}
+
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := s.stateStore.DataAttestorTable().List(
```
Comment (aleem1314) [inline x/data/server/query_attestations_by_hash.go:37]: same here

File: x/data/server/query_attestations_by_iri.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	if err != nil {
+		return nil, sdkerrors.ErrNotFound.Wrap("data record with IRI")
+	}
+
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := s.stateStore.DataAttestorTable().List(
```
Comment (aleem1314) [inline x/data/server/query_attestations_by_iri.go:38]: same

File: x/data/client/query.go [hunk 4]
```diff
@@ -286,16 +304,61 @@ regen q data resolvers-by-iri regen:113gdjFKcVCt13Za6vN7TtbgMM6LMSjRnu89BMCxeuHd
...
+				return err
+			}
+
+			contentHash, err := parseContentHash(ctx, args[0])
+			if err != nil {
+				return err
+			}
+
+			res, err := c.ResolversByHash(cmd.Context(), &data.QueryResolversByHashRequest{
+				ContentHash: contentHash,
```
Comment (aleem1314) [inline x/data/client/query.go:340]: pagination flags.

File: x/data/client/query.go [hunk 5]
```diff
@@ -116,99 +114,117 @@ func QueryByAttestorCmd() *cobra.Command {
...
-			pagination, err := client.ReadPageRequest(cmd.Flags())
+			contentHash, err := parseContentHash(ctx, args[0])
 			if err != nil {
 				return err
 			}
 
-			res, err := c.AttestorsByIRI(cmd.Context(), &data.QueryAttestorsByIRIRequest{
-				Iri:        args[0],
-				Pagination: pagination,
+			res, err := c.AttestationsByHash(cmd.Context(), &data.QueryAttestationsByHashRequest{
```
Comment (aleem1314) [inline x/data/client/query.go:223]: missing pagination request.

File: x/data/server/query_resolvers_by_hash.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	it, err := s.stateStore.DataResolverTable().List(
+		ctx,
+		api.DataResolverPrimaryKey{}.WithId(dataId.Id),
+		ormlist.Paginate(pg),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &data.QueryResolversByHashResponse{}
```
Comment (technicallyty) [inline x/data/server/query_resolvers_by_hash.go:45]:
```suggestion
	}
	defer it.Close()

	res := &data.QueryResolversByHashResponse{}
```

File: x/data/server/query_resolvers_by_iri.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	it, err := s.stateStore.DataResolverTable().List(
+		ctx,
+		api.DataResolverPrimaryKey{}.WithId(dataId.Id),
+		ormlist.Paginate(pg),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &data.QueryResolversByIRIResponse{}
```
Comment (technicallyty) [inline x/data/server/query_resolvers_by_iri.go:46]:
```suggestion
	if err != nil {
		return nil, err
	}
	defer it.Close()

	res := &data.QueryResolversByIRIResponse{}
```

File: x/data/server/query_resolvers_by_url.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	it, err := s.stateStore.ResolverTable().List(
+		ctx,
+		api.ResolverUrlIndexKey{}.WithUrl(request.Url),
+		ormlist.Paginate(pg),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &data.QueryResolversByURLResponse{}
```
Comment (technicallyty) [inline x/data/server/query_resolvers_by_url.go:35]:
```suggestion
	}
	defer it.Close()

	res := &data.QueryResolversByURLResponse{}
```

**Review by technicallyty — APPROVED** [review]
pre-approving. just needs iterator closes 👍🏻

File: x/data/server/query_attestations_by_hash.go (hunk 2 above)
Comment (ryanchristo) [reply x/data/server/query_attestations_by_hash.go:37]: https://github.com/regen-network/regen-ledger/pull/1156/commits/be6dc3ed87a2c8687301c5425f649af1afc1ea23

File: x/data/server/query_attestations_by_iri.go (hunk 3 above)
Comment (ryanchristo) [reply x/data/server/query_attestations_by_iri.go:38]: https://github.com/regen-network/regen-ledger/pull/1156/commits/be6dc3ed87a2c8687301c5425f649af1afc1ea23

File: x/data/client/query.go (hunk 4 above)
Comment (ryanchristo) [reply x/data/client/query.go:352]: https://github.com/regen-network/regen-ledger/pull/1156/commits/be6dc3ed87a2c8687301c5425f649af1afc1ea23

File: x/data/client/query.go (hunk 5 above)
Comment (ryanchristo) [reply x/data/client/query.go:228]: https://github.com/regen-network/regen-ledger/pull/1156/commits/be6dc3ed87a2c8687301c5425f649af1afc1ea23

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**